www/js/diary/diaryHelper.ts

```diff
@@ -109,4 +109,52 @@ export function getFormattedTimeRange(beginFmtTime: string, endFmtTime: string)
   return endMoment.to(beginMoment, true);
 };
 
-// the rest is TODO, still in services.js
+// Temporary function to avoid repear in getPercentages ret val.
+const filterRunning = (mode) =>
+  (mode == 'MotionTypes.RUNNING') ? 'MotionTypes.WALKING' : mode;
+
+export function getPercentages(trip) {
+  if (!trip.sections?.length) return {};
+
+  // sum up the distances for each mode, as well as the total distance
+  let totalDist = 0;
+  const dists: Record<string, number> = {};
+  trip.sections.forEach((section) => {
+    const filteredMode = filterRunning(section.sensed_mode_str);
+    dists[filteredMode] = (dists[filteredMode] || 0) + section.distance;
+    totalDist += section.distance;
+  });
+
+  // sort modes by the distance traveled (descending)
+  const sortedKeys = Object.entries(dists).sort((a, b) => b[1] - a[1]).map(e => e[0]);
+  let sectionPcts = sortedKeys.map(function (mode) {
+    const fract = dists[mode] / totalDist;
+    return {
+      mode: mode,
+      icon: motionTypeOf(mode)?.icon,
+      color: motionTypeOf(mode)?.color || 'black',
+      pct: Math.round(fract * 100) || '<1' // if rounds to 0%, show <1%
+    };
+  });
```

Comment:
can't you just use inferred/cleaned section summary here?

Reply:
I am not sure what is meant by "inferred/cleaned section summary". Can you clarify?

Reply:
As part of the changes prepping for moving "count every trip" into production, we added `cleaned_summary` and `inferred_summary` objects to the `confirmed`, and thus, the composite trips.

From `emission/core/wrapper/confirmedtrip.py`.

```python
 20                   "inferred_section_summary": ecwb.WrapperBase.Access.WORM,
 21                   "cleaned_section_summary": ecwb.WrapperBase.Access.WORM,
```

These contains the percentages per mode of the distance, duration, count, ...
It seems like you can use them directly instead of recomputing